### PR TITLE
feature(product) add delete endpoint

### PR DESCRIPTION
**Pull request by rprzedzik**
# Description

Add delete attribute from product endpoint

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

- [ ] Unit Tests
- [ ] Behat Test

# Checklist:
- [ ] I have read the contribution requirements and fulfill them.
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: module/product/src/Application/Controller/Api/Attribute/DeleteProductsAttributesAction.php [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+        if ($form->isSubmitted() && $form->isValid()) {
+            /** @var DeleteProductAttributeCollectionFormModel $model */
+            $model = $form->getData();
+            foreach ($model->data as $product) {
+                $command = $this->commandFactory->create($product);
+                $this->commandBus->dispatch($command);
+            }
+
+            return new SuccessResponse();
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/DeleteProductsAttributesAction.php:106]: This as far as I see returns 200 HTTP status code, not 204(which is the correct one) as claimed in the documentation.

File: module/product/src/Application/Controller/Api/Attribute/DeleteProductsAttributesAction.php [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        }
+
+
+
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/DeleteProductsAttributesAction.php:110]: So many empty lines seem redundant

File: module/product/src/Application/Controller/Api/Attribute/UpdateProductsAttributesAction.php [hunk 3]
```diff
@@ -72,14 +72,13 @@ public function __construct(
...
-     *     description="Add category ID",
      *     required=true,
-     *     @SWG\Schema(ref="#/definitions/product_category")
+     *     @SWG\Schema(ref="#/definitions/attribute")
      * )
      *
      * @SWG\Response(
      *     response=204,
-     *     description="Update mass pproducts attribtes",
+     *     description="Update mas products attribtes",
```
Comment (piotrkreft) [inline module/product/src/Application/Controller/Api/Attribute/UpdateProductsAttributesAction.php:81]:
```suggestion
     *     description="Update mass products attribtes",
```
BTW didn't we use `batch` convention?

**Review by wiewiurdp — APPROVED** [review]

File: module/product/src/Domain/Command/Attribute/RemoveProductAttributesCommand.php [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+use Ergonode\Product\Domain\Command\ProductCommandInterface;
+
+class RemoveProductAttributesCommand implements ProductCommandInterface
+{
+    private ProductId $id;
+
+    private array $attributes;
+
+    /**
+     * @param array     $attributes
```
Comment (piotrkreft) [inline module/product/src/Domain/Command/Attribute/RemoveProductAttributesCommand.php:21]:
```suggestion
     * @param Type[] attributes
```

File: module/product/src/Application/Model/Product/Attribute/Delete/DeleteAttributeValueFormModel.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    /**
+     * @Assert\NotBlank()
+     * @Assert\Uuid(strict=true)
+     */
+    public ?string $id = null;
+
+    /**
+     * @Assert\All({
+     *     @Assert\NotBlank(),
+     *     @Assert\Regex("/^[a-z]{2}_[A-Z]{2}$/")
```
Comment (piotrkreft) [inline module/product/src/Application/Model/Product/Attribute/Delete/DeleteAttributeValueFormModel.php:24]: LanguageConstraint perhaps?

**Review by piotrkreft — CHANGES_REQUESTED** [review]

**Review by piotrkreft — APPROVED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Pull request merged**